### DISCUSSION TITLE
Data source binding with URL parameter not working.

### DISCUSSION THREAD

**Discussion by mattskiduggan** [Budibase/budibase#6486] [Help] [answered]
Hi,

I'm new to Budibase and it looks great. In my tests I'm trying to drive a line chart from a picker with options and I'm running into an issue binding a data source's URL parameter from a picker.  I'm using Budibase v1.0.206 and Firefox v101.0.1. 

The structure of components I'm using is as follows:-

![image](https://user-images.githubusercontent.com/2762066/176170660-9a3d9284-9762-4a5b-9659-dd6e8789bcc3.png)

1. Use two REST data sources - A. one to provide a set of options for a picker, B. another to retrieve data for a line chart using the value chosen from the picker. 
2. The B data source uses a URL like this:- `https://myhost/1.0/observer/history/{{ jobId }}` and I've created a binding called `jobId` with a default value of `xyz`.
3. Both REST queries work ok in the data admin UI and to drive the picker and chart using the default value.  
4. I've got a 'Create' Form component that uses the Schema of the data use used for the Picker. The contains the others, as per the image above. 
5.  For the picker, the Field parameter is set to `observerType` , the Label column uses the `name` property from the REST query response and the Value column uses an `_id` column. If I set the Label parameter to use `{{ Observers Form.Fields.observerType }}` then I can see the correct `_id` being chosen - this is what I want to pass to the query to drive the chart. 
7. For the Chart Data Provider, I've bound the jobId value to `{{ Observers Form.Fields.observerType }}` but upon doing that, the line chart vanishes. If I remove the handlebar, then the default value is used and the chart is displayed again. 
![image](https://user-images.githubusercontent.com/2762066/176171053-da789ea2-87e3-44cf-81c7-f707ff96b46c.png)

I'm pretty sure that it's something that I'm doing wrong but any advice you can give would be welcome. 

thanks,
Matt

**Comment by melohagan** [comment]
Hi @mattskiduggan 

**Edit** Sorry just noticed you're using two REST api data sources. I would check that the **_id** field of the first definitely is what the second API is expecting.

You can also use a paragraph element to print out the value of `{{ Observers Form.Fields.observerType }}` for debugging purposes.

**Comment by mattdugganibm** [comment, marked as answer]
Hi @melohagan , I've just figured it out - it was an issue with the source data and the binding is working fine now!   Thanks for taking a look.
